rules/static_framework_generator.bzl

```diff
@@ -0,0 +1,214 @@
+    installer = ctx.actions.declare_file(subpath + "/installer.sh")
+    ctx.actions.expand_template(
+        template = ctx.file._installer_template,
+        output = installer,
+        substitutions = {
+            "$(files)": " ".join(files),
+            "$(destination)": destination,
+        },
+        is_executable = True,
+    )
+    return installer
+
+def _static_framework_generator_impl(ctx):
+    frameworks = _make_frameworks(ctx)
+    destination = _make_destination(ctx)
+    installer = _make_installer(ctx, ctx.label.name, [framework.path for framework in frameworks], destination)
+    return [
+        DefaultInfo(
+            executable = installer,
+            runfiles = ctx.runfiles(files = frameworks),
+        ),
+    ]
+
+# Making below rule public in order to properly generate its documentation until the
+# https://github.com/bazelbuild/stardoc/issues/27 issue is resolved
+static_framework_generator_test = rule(
+    implementation = _static_framework_generator_impl,
+    doc = """\
+This generator moves the frameworks artifacts produced by Bazel to a location out of the cache.
+It is intended to be used as a tool to ease compatibility with other build systems.
+
+For example, it would be possible to integrate with cocoapods as follows:
+1- Build targets with Bazel
+2- Use this rule to generate frameworks for them
+3- Generate podspec files containing the generated frameworks as vendored_frameworks, so they can be 
+   integrated in a cocoapods setup, substituting their source code counterparts (useful to cut build time)
+""",
+    attrs = {
+        "targets": attr.label_list(
+            mandatory = True,
+            aspects = [_collect_resources_aspect],
+            doc = "The list of targets to use for generating the frameworks",
+        ),
+        "destination_relative_to_workspace": attr.string(
+            mandatory = False,
+            default = "static_framework_generator",
+            doc = "Destination folder for the generated frameworks, relative to the workspace",
+        ),
+        "destination_relative_to_package": attr.string(
+            mandatory = False,
+            doc = "Destination folder for the generated frameworks, relative to the package",
+        ),
+        "_installer_template": attr.label(
+            default = Label("//tools/static_framework_generator:installer.sh"),
+            allow_single_file = ["sh"],
+        ),
+    },
+    # Ideally this rule would not be a test: it would be an executable to be run with `bazel run`.
+    # The problem with `bazel run` is that you can only execute one target per `bazel run` invocation.
+    # See: https://github.com/bazelbuild/bazel/issues/10855
+    #
+    # A workaround is to mark this rule as a test, because `bazel test` allows to execute multiple
+    # test targets in paralel. Running `bazel test` once with multiple targets is much faster than
+    # running `bazel run` with one target multiple times
+    test = True,
```

Comment:
why can't it be a normal build rule, and the user manually copy a zip or directory or something out of `bazel-bin` ?

Reply:
The reason is to avoid `the user manually copy a zip or directory or something out of bazel-bin`, and instead do it directly with the rule. 

My understanding is that in order to copy files out of the sandbox, the `DefaultInfo` provider needs to return an `executable` to copy them, and only `executable` or `test` rules can do that. Is that correct, or is there other way?

Reply:
For the most part, bazel doesn't allow describing no relative outputs in a build rule that isn't in `bazel-out`. To @segiddins point it might be possible to achieve the result of this program in an ad-hoc program that looks at something like `BEP` or even use `BES` to sync build outputs to an external location e.g. for Xcode, this is how xchammer writes bazel artifacts into DERIVED_DATA for Xcode to pick up and then run